src/components/AuthComponents/RouteGuard.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from "react";
+import SignUpModal from "@/components/AuthComponents/SignUpModal";
+import LoginModal from "@/components/AuthComponents/LoginModal";
+import axios, { AxiosError } from "axios";
+import { useTranslation } from "react-i18next";
+import useAlertStore from "@/stores/alertStore";
+
+const RouteGuard = ({ children }: Readonly<{ children: React.ReactNode }>) => {
+  const [isAuth, setAuth] = useState(false);
+  const [isSignup, setIsSignup] = useState(false);
+  const { showAlert } = useAlertStore();
+  const { t } = useTranslation("authentification");
```

Comment:
Typo. I suggest this extensions to help with typos: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker